### PR TITLE
Update MSRV to 1.63

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

**Comment by alex** [comment]
Hmm, the msrv job is failing, but I don't quite understand why.

**Comment by alex** [comment]
Woo! With enough elbow grease, got it working

File: src/impl_/pymodule.rs [hunk 1]
```diff
@@ -5,9 +5,16 @@ use std::{cell::UnsafeCell, marker::PhantomData};
...
+    not(all(windows, Py_LIMITED_API, not(Py_3_10))),
+    not(target_has_atomic = "64"),
 ))]
 use portable_atomic::{AtomicI64, Ordering};
+#[cfg(all(
+    not(any(PyPy, GraalPy)),
+    Py_3_9,
+    not(all(windows, Py_LIMITED_API, not(Py_3_10))),
+))]
+use std::sync::atomic::{AtomicI64, Ordering};
```
Comment (Icxolu) [inline src/impl_/pymodule.rs:17]: Don't we need `target_has_atomic = "64"` here?

Comment (adamreichold) [reply src/impl_/pymodule.rs:17]: I would rather ask why do we want to second guess the same logic which `portable_atomic` should do itself, i.e. `portable_atomic::AtomicI64` already should essentially be an alias for `std::sync::atomic::AtomicI64` if the target supports that. So why do we want to make the dependency conditional?

Comment (alex) [reply src/impl_/pymodule.rs:17]: `portable_atomic` relies on a `build.rs` which are relatively heavy-weight in terms of compile time, so the goal here is to minimize build time.

File: src/gil.rs [hunk 2]
```diff
@@ -770,29 +760,26 @@ mod tests {
...
+            *self.set.lock().unwrap() = true;
             self.wait.notify_all();
         }
 
         fn wait(&self) {
-            let mut set = self.set.lock();
-            while !*set {
-                self.wait.wait(&mut set);
-            }
+            drop(self.wait.wait_while(self.set.lock().unwrap(), |s| !*s));
```
Comment (adamreichold) [inline src/gil.rs:782]: Why `drop` instead of `unwarp` the poisoning error here?

File: src/impl_/pymodule.rs (hunk 1 above)
Comment (adamreichold) [reply src/impl_/pymodule.rs:18]: Agreed, I remember that discussion now and making the build more hermetic is certainly a worthwhile goal.

**Review by adamreichold — APPROVED** [review]
LGTM except the one nit in the `Event` helper.

File: src/gil.rs (hunk 2 above)
Comment (alex) [reply src/gil.rs]: It needed to be `drop(unwrap())`, but good catch.

**Comment by alex** [comment]
Ok, should be good now.

**Pull request merged**